internal/scheduler/nodeiteration_test.go

```diff
@@ -0,0 +1,536 @@
+			0: {Resources: map[string]resource.Quantity{"cpu": resource.MustParse("4"), "memory": resource.MustParse("4Gi")}},
+			1: {Resources: map[string]resource.Quantity{"cpu": resource.MustParse("5"), "memory": resource.MustParse("5Gi")}},
+			2: {Resources: map[string]resource.Quantity{"cpu": resource.MustParse("6"), "memory": resource.MustParse("6Gi")}},
+		},
+	},
+	{
+		Id:         "node3",
+		NodeTypeId: "bar",
+		NodeType:   &schedulerobjects.NodeType{Id: "bar"},
+		AvailableByPriorityAndResource: map[int32]schedulerobjects.ResourceList{
+			0: {Resources: map[string]resource.Quantity{"cpu": resource.MustParse("7"), "memory": resource.MustParse("7Gi")}},
+			1: {Resources: map[string]resource.Quantity{"cpu": resource.MustParse("8"), "memory": resource.MustParse("8Gi")}},
+			2: {Resources: map[string]resource.Quantity{"cpu": resource.MustParse("9"), "memory": resource.MustParse("9Gi")}},
+		},
+	},
+}
+
+// testNodeItems2 returns a randomly generated set of n nodes.
+func testNodeItems2(priorities []int32, resources []string, n int) []*schedulerobjects.Node {
+	rv := make([]*schedulerobjects.Node, n)
+	for i := 0; i < n; i++ {
+		rv[i] = &schedulerobjects.Node{
+			Id:         uuid.NewString(),
+			NodeTypeId: "foo", // All nodes have the same node type.
+			NodeType:   &schedulerobjects.NodeType{Id: "bar"},
+		}
+		availableByPriorityAndResource := schedulerobjects.NewAvailableByPriorityAndResourceType(priorities, nil)
+		for _, p := range priorities {
+			rs := make(map[string]resource.Quantity)
+			for _, r := range resources {
+				rs[r] = resource.MustParse(fmt.Sprintf("%d", rand.Intn(100)))
+			}
+			availableByPriorityAndResource.MarkAvailable(p, schedulerobjects.ResourceList{Resources: rs})
+		}
+		rv[i].AvailableByPriorityAndResource = availableByPriorityAndResource
+	}
+	return rv
+}
+
+func testNodes3(numCpuNodes, numTaintedCpuNodes, numGpuNodes int, priorities []int32) []*schedulerobjects.Node {
+	rv := make([]*schedulerobjects.Node, 0)
+	for i := 0; i < numCpuNodes; i++ {
+		rv = append(rv, testCpuNode(priorities))
+	}
+	for i := 0; i < numTaintedCpuNodes; i++ {
+		rv = append(rv, testTaintedCpuNode(priorities))
+	}
+	for i := 0; i < numGpuNodes; i++ {
+		rv = append(rv, testTaintedGpuNode(priorities))
+	}
+	return rv
+}
+
+func testNCpuNode(n int, priorities []int32) []*schedulerobjects.Node {
+	rv := make([]*schedulerobjects.Node, n)
+	for i := 0; i < n; i++ {
+		rv[i] = testCpuNode(priorities)
+	}
+	return rv
+}
+
+func testNTaintedCpuNode(n int, priorities []int32) []*schedulerobjects.Node {
+	rv := make([]*schedulerobjects.Node, n)
+	for i := 0; i < n; i++ {
+		rv[i] = testTaintedCpuNode(priorities)
+	}
+	return rv
+}
+
+func testNTaintedtGpuNode(n int, priorities []int32) []*schedulerobjects.Node {
+	rv := make([]*schedulerobjects.Node, n)
+	for i := 0; i < n; i++ {
+		rv[i] = testTaintedGpuNode(priorities)
+	}
+	return rv
+}
+
+// func testNA100Node(n int, priorities []int32) []*schedulerobjects.Node {
```

Comment:
remove?

Reply:
I want to add tests using these. So I'd prefer to keep them for now.